### PR TITLE
[MINOR][SQL][PYTHON] Rename Python plan node names

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR is a sort of a followup of https://github.com/apache/spark/pull/38624 that proposes to rename the plan nodes for Python as below:

From:

```
package org.apache.spark.sql.execution.python

MapInBatchExec
├── MapInPandasExec
└── *PythonMapInArrowExec* (and *PythonMapInArrow*)

*FlatMapCoGroupsInPythonExec*
├── FlatMapCoGroupsInArrowExec
└── FlatMapCoGroupsInPandasExec

*FlatMapGroupsInPythonExec*
├── FlatMapGroupsInArrowExec
└── FlatMapGroupsInPandasExec
```

To:

```
package org.apache.spark.sql.execution.python

MapInBatchExec
├── MapInPandasExec
└── *MapInArrowExec* (and *MapInArrow*)

*FlatMapCoGroupsInBatchExec*
├── FlatMapCoGroupsInArrowExec
└── FlatMapCoGroupsInPandasExec

*FlatMapGroupsInBatchExec*
├── FlatMapGroupsInArrowExec
└── FlatMapGroupsInPandasExec
```

### Why are the changes needed?

To have the consistent names for Python related execution nodes.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing CI should pass.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Build: https://github.com/HyukjinKwon/spark/actions/runs/7225644041

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala [hunk 1]
```diff
@@ -839,7 +839,7 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       case logical.MapInPandas(func, output, child, isBarrier) =>
         execution.python.MapInPandasExec(func, output, planLater(child), isBarrier) :: Nil
       case logical.PythonMapInArrow(func, output, child, isBarrier) =>
```
Comment (ueshin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala]: Do we also want to rename logical plans? I see at least `logical.PythonMapInArrow` here, which is inconsistent with the physical plan.

**Review by ueshin — APPROVED** [review]
Otherwise, LGTM.

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 2]
```diff
@@ -3554,7 +3554,7 @@ class Dataset[T] private[sql](
   private[sql] def pythonMapInArrow(func: PythonUDF, isBarrier: Boolean = false): DataFrame = {
```
Comment (zhengruifeng) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:3554]: shall we rename this `pythonMapInArrow` as well? or we can make it in separate pr

**Review by allisonwang-db — APPROVED** [review]
Now the plan name makes more sense

**Review by zhengruifeng — APPROVED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala (hunk 2 above)
Comment (HyukjinKwon) [reply sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:3554]: LGTM

**Comment by HyukjinKwon** [comment]
Merged to master.
